elements/core/components/Button/Button.tsx

```diff
@@ -9,25 +9,42 @@ interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
   small?: boolean,
   reset?: boolean,
   icon?: IconProp,
+  iconRight?: boolean,
 }
 
 const ButtonComponent = ({
   children,
   disabled,
   type = 'button',
   icon,
+  iconRight,
   ...rest
 }: ButtonProps) => (
   <Button
     icon={icon !== undefined}
     type={type}
+    iconRight={iconRight}
     disabled={disabled}
     {...rest}
   >
-    {icon && (
-      <Icon icon={icon} />
-    )}
-    { children}
+    {
```

Comment:
when you change the styling like in the comment above, you can get rid of the ternary. und just pass the prop